src/role_extractor.js

```diff
@@ -21,10 +21,58 @@ roles.extractor.energyBuild = function(room, energy, heal) {
   return energy;
 };
 
-roles.extractor.action = function(creep) {
-  return creep.handleExractor();
-};
+roles.extractor.terminalStorageExchange = function(creep) {
+  var terminal = creep.room.terminal;
+  if (!terminal || !terminal.isActive()) {
+    // TODO kill creep?
```

Comment:
Killing the creep is fine.

The `isActive()` method is somehow expensive, could be fine for just the mineral roles